tests/unit_tests/cli/test_integration_cli.py

```diff
@@ -403,12 +422,17 @@ def test_that_prior_is_not_overwritten_in_ensemble_experiment(
                 "--port-range",
                 "1024-65535",
                 "--realizations",
-                "0-4",
+                reals_rerun_option,
             ],
         )
 
         FeatureToggling.update_from_args(parsed)
         run_cli(parsed)
         post_facade = LibresFacade.from_config_file("poly.ert")
         parameter_values = post_facade.load_all_gen_kw_data("default")
-        pd.testing.assert_frame_equal(parameter_values, prior_values)
+
+        if should_resample:
```

Comment:
Could also do: 
```python
from contextlib import ExitStack as does_not_raise

with expectation:
    ...
```
where:
```python
exception = does_not_raise()
``` 
or:
```python
exception = pytest.raises(
                AssertionError, match="my match"
            ),
```

Reply:
So:

```python
@pytest.mark.integration_test
@pytest.mark.parametrize(
    "prior_mask,reals_rerun_option,expectation",
    [
        pytest.param(
            None,
            "0-4",
            does_not_raise(),
            id="All realisations first, subset second run",
        ),
        pytest.param(
            [True] * 3,
            "0-5",
            pytest.raises(AssertionError),
            id="Subset of realisation first, superset in second run - must resample",
        ),
    ],
)
def test_that_prior_is_not_overwritten_in_ensemble_experiment(
    prior_mask, reals_rerun_option, expectation, tmpdir, source_root, capsys
):
    with tmpdir.as_cwd():
        ert = EnKFMain(ErtConfig.from_file("poly_example/poly.ert"))
        ....

        with expectation:
            pd.testing.assert_frame_equal(parameter_values, prior_values)
```
...?

You're loosing the context of why it should fail on assert though?